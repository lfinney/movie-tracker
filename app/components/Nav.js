import React from 'react';
import { Link } from 'react-router-dom';

const Nav = ({ isLoggedIn, signOut }) => {
  return (
    <div id="nav-div">
      <Link to='/' className="left">
        Home
      </Link>
      <Link to='/favorites' className="left">
        Favorites
      </Link>
      <Link to='/login' className="right">
        Login
      </Link>

      <button onClick={signOut} className="right">Sign Out</button>
    </div>
  );
};

export default Nav;
