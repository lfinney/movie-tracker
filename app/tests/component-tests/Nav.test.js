import React from 'react';
import { shallow } from 'enzyme';
import Nav from '../../components/Nav';

describe('Nav', () => {

  it('should render component', () => {
    const wrapper = shallow(<Nav />);

    expect(wrapper).toMatchSnapshot();
  });
});
