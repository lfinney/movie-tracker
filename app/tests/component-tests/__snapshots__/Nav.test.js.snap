// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should render component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Nav />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Link
          replace={false}
          to="/"
        >
          Home
        </Link>,
        <Link
          replace={false}
          to="/login"
        >
          Login
        </Link>,
        <Link
          replace={false}
          to="/favorites"
        >
          Favorites
        </Link>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": "Home",
          "replace": false,
          "to": "/",
        },
        "ref": null,
        "rendered": "Home",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": "Login",
          "replace": false,
          "to": "/login",
        },
        "ref": null,
        "rendered": "Login",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": "Favorites",
          "replace": false,
          "to": "/favorites",
        },
        "ref": null,
        "rendered": "Favorites",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Link
            replace={false}
            to="/"
          >
            Home
          </Link>,
          <Link
            replace={false}
            to="/login"
          >
            Login
          </Link>,
          <Link
            replace={false}
            to="/favorites"
          >
            Favorites
          </Link>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": "Home",
            "replace": false,
            "to": "/",
          },
          "ref": null,
          "rendered": "Home",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": "Login",
            "replace": false,
            "to": "/login",
          },
          "ref": null,
          "rendered": "Login",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": "Favorites",
            "replace": false,
            "to": "/favorites",
          },
          "ref": null,
          "rendered": "Favorites",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
